File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        set<int> set;
        vector<int> ans(2);
        for(int i:nums){
            if(set.find(i)!=set.end()){
                ans[0]=i;
            }
            set.insert(i);
        }
        for(int i=1;i<=nums.size();i++){
            if(set.find(i)==set.end()){
                ans[1]=i;
                break;
            }
        }
        return ans;
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        HashSet<Integer> set=new HashSet<>();
        int[] ans=new int[2];
        for(int i:nums){
            if(set.contains(i)){
                ans[0]=i;
            }
            set.add(i);
        }
        for(int i=1;i<=nums.length;i++){
            if(!set.contains(i)){
                ans[1]=i;
                break;
            }
        }
        return ans;
    }
}